### PR TITLE
PLATUI-3297 fix bug with service navigation twirl

### PR DESCRIPTION
**Pull request by TimothyFothergill**
# Purpose of PR
- To fix a bug with the `ServiceNavigation` component twirl template
  - This addresses a discrepancy between our `twirl` template and the `govuk-frontend` `nunjucks` equivalent
  - The slotted content would always provide a `nav` element if any slot was filled, but we only want the `nav` element if a `navigationStart` or `navigationEnd` slot has been filled with content
- Add a new test and update an existing test around this edge case

### CODE REVIEW

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukServiceNavigationSpec.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ class GovukServiceNavigationSpec
       output.first().attr("href") shouldBe "#"
     }
 
-    "render ServiceNavigation with slotted content" in {
+    "render ServiceNavigation with slotted content out of the nav element" in {
```
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukServiceNavigationSpec.scala:73]: what about renaming this something like

"render the nav element even if there are no nav items when either of the navigation slots are provided"

**Review by oscarduignan — APPROVED** [review]

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukServiceNavigationSpec.scala (hunk 1 above)
Comment (TimothyFothergill) [reply play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukServiceNavigationSpec.scala:73]: This test ensures we do **not** render the nav element, as neither the navigation nor a navigation Slot is added. Should I add a third test for when we have a nav element without it being slotted content?

**Review by JoPintoPaul — APPROVED** [review]
